install/migrations/update_10.0.x_to_11.0.0/assets.php

```diff
@@ -179,6 +179,40 @@
     $DB->doQueryOrDie($query);
 }
 
+if (!$DB->tableExists('glpi_assets_customfields')) {
+    $query = <<<SQL
+        CREATE TABLE `glpi_assets_customfields` (
+          `id` int {$default_key_sign} NOT NULL AUTO_INCREMENT,
+          `assets_assetdefinitions_id` int {$default_key_sign} NOT NULL,
+          `name` varchar(255) NOT NULL,
+          `label` varchar(255) NOT NULL,
+          `type` varchar(255) NOT NULL,
+          `flags` tinyint NOT NULL DEFAULT 0,
+          `itemtype` VARCHAR(255) NULL DEFAULT NULL,
+          `default_value` text,
+          `allowed_values` text,
+          PRIMARY KEY (`id`),
+          UNIQUE KEY `unicity` (`assets_assetdefinitions_id`, `name`)
+        ) ENGINE=InnoDB DEFAULT CHARSET={$default_charset} COLLATE={$default_collation} ROW_FORMAT=DYNAMIC;
+SQL;
+    $DB->doQueryOrDie($query);
+}
+
+if (!$DB->tableExists('glpi_assets_customfieldvalues')) {
+    $query = <<<SQL
+        CREATE TABLE `glpi_assets_customfieldvalues` (
+          `id` int {$default_key_sign} NOT NULL AUTO_INCREMENT,
+          `assets_customfields_id` int {$default_key_sign} NOT NULL,
+          `assets_assets_id` int {$default_key_sign} NOT NULL,
+          `value` text,
+          PRIMARY KEY (`id`),
+          UNIQUE KEY `unicity` (`assets_customfields_id`, `assets_assets_id`),
+          KEY `value` (`value`(255))
+        ) ENGINE=InnoDB DEFAULT CHARSET={$default_charset} COLLATE={$default_collation} ROW_FORMAT=DYNAMIC;
+SQL;
+    $DB->doQueryOrDie($query);
+}
```

Comment:
The EAV solution will be a problem due to the max join limitation in MySQL. Indeed, it will require a join operation for each displayed column, and the limit could be reach too quickly.

A JSON field to store all the values of the item would be preferable.